src/test/java/org/jenkinsci/plugins/workflow/cps/global/CompilationErrorExceptionTest.java

```diff
@@ -0,0 +1,42 @@
+package org.jenkinsci.plugins.workflow.cps.global;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import hudson.model.Run;
+import hudson.plugins.git.GitSCM;
+import jenkins.plugins.git.GitSCMSource;
+import jenkins.plugins.git.GitSampleRepoRule;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.libs.GlobalLibraries;
+import org.jenkinsci.plugins.workflow.libs.LibraryConfiguration;
+import org.jenkinsci.plugins.workflow.libs.SCMRetriever;
+import org.jenkinsci.plugins.workflow.libs.SCMSourceRetriever;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.Collections;
+
+@Issue("JENKINS-40109")
+public class CompilationErrorExceptionTest extends Assert {
```

Comment:
nit: Maybe better to just create a file `UserDefinedGlobalVariable.java` in case we want to add more tests for that class, but I think things are fine as-is.

Reply:
I say, let's leave it as is. It can always be renamed later if needed.